### PR TITLE
DX: check that tests are agnostic of PHP version

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

File: tests/AutoReview/ProjectCodeTest.php [hunk 1]
```diff
@@ -354,6 +354,41 @@ public function testThatTestClassCoversAreCorrect(string $testClassName): void
         }
     }
 
+    /**
+     * @dataProvider provideTestClassCases
+     */
+    public function testThereIsNoPhpVersionUsedDirectly(string $className): void
```
Comment (keradus) [inline tests/AutoReview/ProjectCodeTest.php:360]: putting on hold for now. It's not a decision to "reject this idea". It's to indicate we need to explicitly decide on approach and I rather toward not blocking usage of PHP_ in whole test file.

same as https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7580#pullrequestreview-1785318277

**Review by keradus — CHANGES_REQUESTED** [review]
(preventing accidental merge - on hold, decision needed)

File: tests/AutoReview/ProjectCodeTest.php (hunk 1 above)
Comment (kubawerlos) [reply tests/AutoReview/ProjectCodeTest.php:360]: > I rather toward not blocking usage of PHP_ in whole test file.

Why? Let's be pragmatic, changes in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7581 and in here show that `PHP_*` token is only used unnecessarily.

Why not block it everywhere and if any usage will be reasonable (which I doubt will ever happen as it has not happen yet) make amends?

Blocking it only in data providers or `if` conditions will not help to protect before our main concern, as a simple workaround will be to extract the check to a method like this:
```
private function shouldTestAttributes(): bool
{
    return \PHP_VERSION_ID >= 8_00_00;
}
```

Comment (keradus) [reply tests/AutoReview/ProjectCodeTest.php:360]: as mentioned in previous discussion, i explicitly wanted to _NOT_ decide if we check in whole file or only dataProviders _before_ we clean up the tests, simply to see how many cases outside of dataProviders we will have, if any.
I prefer to made the decision based on data and not gut feelings before implementation.

if we cleaned up all tests, we can decide now ;)

Comment (kubawerlos) [reply tests/AutoReview/ProjectCodeTest.php:360]: > I prefer to made the decision based on data and not gut feelings

Me too, that is why I did `git grep PHP_VERSION_ID ./tests/Fixer` and the result was quite small to analyse.

**Pull request closed without merging**